crates/sqllib/src/string.rs

```diff
@@ -351,38 +383,109 @@ some_function3!(
     String
 );
 
+#[doc(hidden)]
 pub fn writelog<T: std::fmt::Display>(format: String, argument: T) -> T {
     let format_arg = format!("{}", argument);
     let formatted = format.replace("%%", &format_arg);
     print!("{}", formatted);
     argument
 }
 
+#[doc(hidden)]
 pub fn parse_json_s(value: String) -> Variant {
     match serde_json::from_str::<Variant>(&value) {
         Ok(v) => v,
         Err(_) => Variant::SqlNull,
     }
 }
 
+#[doc(hidden)]
 pub fn parse_json_nullN(_value: Option<()>) -> Option<Variant> {
     None
 }
 
 some_polymorphic_function1!(parse_json, s, String, Variant);
 
+#[doc(hidden)]
 pub fn to_json_V(value: Variant) -> Option<String> {
     match value.to_json_string() {
         Ok(s) => Some(s),
         _ => None,
     }
 }
 
+#[doc(hidden)]
 pub fn to_json_VN(value: Option<Variant>) -> Option<String> {
     let value = value?;
     to_json_V(value)
 }
 
+#[doc(hidden)]
 pub fn to_json_nullN(_value: Option<()>) -> Option<String> {
     None
 }
+
+#[doc(hidden)]
+pub fn regexp_replace3___(str: String, re: String, repl: String) -> String {
```

Comment:
Compiling the regex every time is _really_ expensive (I've tried). I think the right solution is to cache them using `re` string as the key (also tried, and it's orders of magnitude more efficient). In practice most programs only use a few fixed regular expressions, so the cache will remain small. But in case it's not, we need [LRU](https://crates.io/search?q=lru).

Reply:
they are cached if the string is a constant.
see regexp_replaceC3___

Reply:
the compiler will call this function only if the regular expression is not a constant string.

Reply:
Ok, that's clever. I still think caching would be useful though.

Reply:
I agree, we can file a more general issue about this.
GC for that cache is an open question.

Reply:
Pretty sure, LRU with a cache of a few thousand entries will be good enough